scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+                                kwargs=kwargs, log=True)
+
+    def cdf(self, x, y=None, *, method=None):
+        r"""Cumulative distribution function
+
+        The cumulative distribution function :math:`F(x)` is the probability
+        the random variable :math:`X` will take on a value less than or equal
+        to :math:`x`:
+
+        .. math::
+
+            F(x) = P(X ≤ x)
+
+        A two-argument variant of this function is also defined as the
+        probability the random variable :math:`X` will take on a value between
+        :math:`x` and :math:`y`.
+
+        .. math::
+
+            F(x, y) = P(x ≤ X ≤ y)
+
+        ``cdf`` accepts `x` for :math:`x` and `y` for :math:`y`.
+
+        Parameters
+        ----------
+        x, y : Array
+            The arguments of the cumulative distribution function (CDF). `x` is
+            required; `y` is optional.
+        method : {None, 'formula', 'logexp', 'complementarity', 'quadrature', 'subtraction'}
+            The strategy used to evaluate the CDF.
+            By default (``None``), the one-argument form of the function
+            chooses between the following options.
+
+            - ``'formula'``: use a formula for the CDF itself
+            - ``'logexp'``: evaluate the logarithm of the CDF directly and
+                            exponentiate
+            - ``'complementarity'``: evaluate the complementary CDF direcly
+                                     and take the complement
+            - ``'quadrature'``: numerically integrate the PDF
+
+            In place of ``'complementarity'``, the two-argument form accepts:
+
+            - ``'subtraction'``: compute the CDF at each argument and take
+              the difference.
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The CDF evaluated at the provided argument(s).
+
+        See Also
+        --------
+        ContinuousDistribution.logcdf
+        ContinuousDistribution.ccdf
+
+        Notes
+        -----
+        Suppose a probability distribution has support :math:`[l, r]`.
+        The CDF :math:`F(x)` is related to the probability density function
+        :math:`f(x)` by:
+
+        .. math::
+
+            F(b) = \int_l^b f(x) dx
+
+        The two argument version is:
+
+        .. math::
+
+            F(a, b) = \int_a^b f(x) dx = F(b) - F(a)
+
+        The CDF evaluates to its minimum value of :math:`0` for :math:`x ≤ l`
+        and its maximum value of :math:`1` for :math:`x ≥ r`.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-0.5, b=0.5)
+
+        Evaluate the CDF at the desired argument:
+
+        >>> X.cdf(0.25)
+        0.75
+
```

Comment:
an example like X.cdf(0.1, 0.2) == X.cdf(0.4, 0.5) could be helpful to explain the two-argument version

Reply:
Yes, an example was needed. Added.

Reply:
can be resolved